server/src/main/java/org/opensearch/cluster/routing/remote/InternalRemoteRoutingTableService.java

```diff
@@ -53,12 +56,13 @@
  * @opensearch.internal
  */
 public class InternalRemoteRoutingTableService extends AbstractLifecycleComponent implements RemoteRoutingTableService {
-
+    
     private static final Logger logger = LogManager.getLogger(InternalRemoteRoutingTableService.class);
     private final Settings settings;
     private final Supplier<RepositoriesService> repositoriesService;
     private Compressor compressor;
     private RemoteWritableEntityStore<IndexRoutingTable, RemoteIndexRoutingTable> remoteIndexRoutingTableStore;
+    private RemoteWritableEntityStore<RemoteIndexRoutingTableDiff, RemoteIndexRoutingTableDiff> remoteIndexRoutingTableDiffStore;
```

Comment:
We need to map non-remote to remote entity here, not sure if this signature makes sense

Reply:
Created `RoutingTableIncrementalDiff` non-remote entity.